spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/microsoft/teams/MicrosoftTeamsNotifierTest.java

```diff
@@ -0,0 +1,231 @@
+
+        mockRestTemplate = mock(RestTemplate.class);
+        testMTNotifier.setRestTemplate(mockRestTemplate);
+        testMTNotifier.setWebhookUrl(webhookUrl);
+
+        stubApp = Application.create(appName)
+                .withId(appId)
+                .withHealthUrl(healthUrl)
+                .withManagementUrl(managementUrl)
+                .withServiceUrl(serviceUrl).build();
+    }
+
+    @Test
+    public void test_onClientApplicationDeRegisteredEvent_resolve() throws Exception{
+        ClientApplicationDeregisteredEvent deregisteredEvent = new ClientApplicationDeregisteredEvent(stubApp);
+
+        Object expected = testMTNotifier.getDeregisteredMessage(stubApp);
+
+        testMTNotifier.doNotify(deregisteredEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationRegisteredEvent_resolve() throws Exception {
+        ClientApplicationRegisteredEvent registeredEvent = new ClientApplicationRegisteredEvent(stubApp);
+
+        Object expected = testMTNotifier.getRegisteredMessage(stubApp);
+
+        testMTNotifier.doNotify(registeredEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromDownToUp_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofDown(), StatusInfo.ofUp());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofDown(), StatusInfo.ofUp());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromUpToDown_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofUp(), StatusInfo.ofDown());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofUp(), StatusInfo.ofDown());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromUpToOffline_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofUp(), StatusInfo.ofOffline());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofUp(), StatusInfo.ofOffline());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
```

Comment:
What do you want to assert here? The `expected` message is produced by the test under class itself.
So it doesn't make sense to asser equality here. If you just wanted to make sure that the restTemplate is called, assert for any(Message.class). Maybe it is worth to assert the url.